test/e2e/nodes_test.go

```diff
@@ -5,17 +5,36 @@ import (
 	. "github.com/onsi/gomega"
 
 	"k8s.io/apimachinery/pkg/types"
+
+	yaml "sigs.k8s.io/yaml"
+
+	nmstatev1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1"
 )
 
 var _ = Describe("Nodes", func() {
 	Context("when nodes are up", func() {
 		It("should have NodeNetworkState with currentState for each node", func() {
 			for _, node := range nodes {
 				key := types.NamespacedName{Namespace: namespace, Name: node}
-				nodeNetworkState := nodeNetworkState(key)
-				//TODO: exec nmstatectl is not in place let's just compare
-				//      with the stuff we have harcoded there
-				Expect(nodeNetworkState.Status.CurrentState).To(MatchYAML("interfaces: []"))
+				var currentStateYaml nmstatev1.State
+				Eventually(func() nmstatev1.State {
+					currentStateYaml = nodeNetworkState(key).Status.CurrentState
+					return currentStateYaml
+				}).ShouldNot(BeEmpty(), "Node %s should have currentState", node)
+
+				By("unmarshal state yaml into unstructured golang")
+				var currentState map[string]interface{}
+				err := yaml.Unmarshal(currentStateYaml, &currentState)
+				Expect(err).ToNot(HaveOccurred(), "should parse correctly yaml: %s", currentStateYaml)
```

Comment:
Other messages start with capital -> Should

Reply:
Done